core/input_map.cpp

```diff
@@ -175,12 +175,92 @@ bool InputMap::event_is_action(const Ref<InputEvent> &p_event, const StringName
 		ERR_FAIL_COND_V(!E, false);
 	}
 
-	Ref<InputEventAction> iea = p_event;
-	if (iea.is_valid()) {
-		return iea->get_action() == p_action;
+	Ref<InputEventAction> input_event_action = p_event;
+	if (input_event_action.is_valid()) {
+		return input_event_action->get_action() == p_action;
 	}
 
-	return _find_event(E->get().inputs, p_event, true) != NULL;
+	return _find_event(E->get().inputs, p_event) != NULL;
```

Comment:
There is no need to add 3 identical functions here, just add optional arguments to event_is_action_pressed such as float *r_strength=NULL, bool *r_pressed=NULL